pallet/staking/src/lib.rs

```diff
@@ -373,18 +382,13 @@ pub mod pallet {
 				} else {
 					<frame_system::Pallet<T>>::inc_consumers(&who)?;
 
-					*l = Some(Ledger {
-						staked_ring: Default::default(),
-						staked_deposits: Default::default(),
-						unstaking_ring: Default::default(),
-						unstaking_deposits: Default::default(),
-					});
+					*l = Some(Ledger { ring: Default::default(), deposits: Default::default() });
```

Comment:
Also need rate limiting when stake

Reply:
Ok, will set up a separate storage to track incoming stakes.

Reply:
I think it is OK to use the same config

Reply:
Sure, I just came up with an idea.

```rs
enum Accumulator {
    Pos(u128),
    Neg(u128),
}
```

Stake will reduce the negative part, then increase the positive part.
The opposite is true for unstaking.

Reply:
Do not allow the positive or negative part to exceed 10,000,000,000, totaling 20,000,000,000 as per your request.

Reply:
> Sure, I just came up with an idea.
> 
> 
> 
> ```rs
> 
> enum Accumulator {
> 
>     Pos(u128),
> 
>     Neg(u128),
> 
> }
> 
> ```
> 
> 
> 
> Stake will reduce the negative part, then increase the positive part.
> 
> The opposite is true for unstaking.

This make more outgoing unstake can be done by incoming stake.

Not quite sure about it, looks more complex for rate limiting.

Reply:
This seems more like a dynamic rate limiting strategy. It does introduce a bit more extra work, but that's acceptable to me.

Which one do you prefer?

Reply:
It looks good to me after rethinking.

Reply:
I still prefer simple one using same config, but dynamic one looks good to me, you decide it.